cadvisor/getmetrics_cadvisor.py

```diff
@@ -24,12 +25,15 @@
 hostname = socket.gethostname()
 cAdvisoraddress = "http://" + hostname + ":8080/api/v1.3/docker/"
 
-command = ['bash', '-c', 'source ' + str(homepath) + '/.agent.bashrc && env']
-proc = subprocess.Popen(command, stdout=subprocess.PIPE)
-for line in proc.stdout:
-    (key, _, value) = line.partition("=")
-    os.environ[key] = value.strip()
-proc.communicate()
+try:
+    if os.path.exists(os.path.join(homepath, "cadvisor", "config.ini")):
+        parser = SafeConfigParser()
+        parser.read(os.path.join(homepath, "cadvisor", "config.ini"))
+        sampling_interval = parser.get('cadvisor', 'sampling_interval')
+        os.environ["SAMPLING_INTERVAL"] = sampling_interval
```

Comment:
environment variable setting can be ommitted/may be not needed. Refer other newly refactored agents agents